services/data-graph/src/main/validate-data-graph-message.ts

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { DataGraphMessage } from '@carbon-platform/api/data-graph'
+import { UnvalidatedMessage } from '@carbon-platform/api/messaging'
+import { InvalidInputException } from '@carbon-platform/api/microservice'
+
+function validateDataGraphMessage(data: UnvalidatedMessage): DataGraphMessage {
```

Comment:
curious what would happen if the query was a valid string and the variables was a valid object but it didn't contain a key that was needed? 🤔

Reply:
You can kinda think of this as the first level of message validation (one of two). This one makes sure that the incoming message from the broker has the expected properties based on its interface.

The second one is the actual invocation of `graphql(...)` over in https://github.com/carbon-design-system/carbon-platform/pull/720/files#diff-ca57878b44a9309813e113add66fff05e4f11acd3f58cbec1bda4ea7a87d3f1dR49

That's the one that takes the actual graphql schema as input and validates the incoming query. If it's malformed, that will generate errors which will end up in the `{errors: ...}` section of the query response.